blob/blob.go

```diff
@@ -638,12 +644,18 @@ func (b *Bucket) NewRangeReader(ctx context.Context, key string, offset, length
 			b.tracer.End(tctx, err)
 		}
 	}()
-	r, err := b.b.NewRangeReader(ctx, key, offset, length, dopts)
+	dr, err := b.b.NewRangeReader(ctx, key, offset, length, dopts)
 	if err != nil {
 		return nil, wrapError(b.b, err)
 	}
 	end := func(err error) { b.tracer.End(tctx, err) }
-	return &Reader{b: b.b, r: r, end: end, provider: b.tracer.Provider}, nil
+	r := &Reader{b: b.b, r: dr, end: end, provider: b.tracer.Provider}
+	runtime.SetFinalizer(r, func(r *Reader) {
+		if !r.closed {
+			log.Printf("A blob.Reader reading from %q was never closed", key)
```

Comment:
To help the user even further in debugging, you can include the caller of `NewRangeReader`. Snippet to get file/lineno which may or may not be available:

```go
// Call this inside `NewRangeReader`, not the finalizer:
_, file, lineno, ok := runtime.Caller(1)
```

You may also need to tweak `NewReader`, since that will be a common caller of `NewRangeReader`.

(You should also do this for `NewWriter`.)

Reply:
Done.